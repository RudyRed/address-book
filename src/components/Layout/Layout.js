import React from 'react';
import { Typography, withStyles } from '@material-ui/core';

const Layout = ({ children, classes, title }) => {
  return (
    <div className={classes.root}>
      <div className={classes.header}>
        <Typography variant="h1">{title}</Typography>
      </div>
      <main className={classes.flex}>{children}</main>
    </div>
  );
};

const styles = (theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    marginLeft: 'auto',
    marginRight: 'auto',
    maxWidth: 750,
    minHeight: '100vh',
  },
  header: {
    margin: theme.spacing(2, 0, 1),
  },
  headerContainer: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  flex: { flex: 1 },
});

export default withStyles(styles, { withTheme: true })(Layout);
