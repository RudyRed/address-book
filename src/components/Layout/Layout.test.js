import { render } from '@testing-library/react';
import { Typography } from '@material-ui/core';

import Layout from './Layout';

const childrenText = 'Layout-children-text';
const defaultProps = {
  children: <Typography>{childrenText}</Typography>,
  title: 'Layout-title',
};

const setup = (overrides) => {
  const props = {
    ...defaultProps,
    ...overrides,
  };
  return {
    ...render(<Layout {...props} />),
    props,
  };
};

describe('Layout', () => {
  it('Should render Layout', () => {
    const { getByText, props } = setup();

    expect(getByText(props.title)).toBeInTheDocument();
    expect(getByText(childrenText)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
