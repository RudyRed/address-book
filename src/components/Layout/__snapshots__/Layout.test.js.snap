// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<div>
  <div
    class="Layout-root-5"
  >
    <div
      class="Layout-header-6"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        Layout-title
      </h1>
    </div>
    <main
      class="Layout-flex-8"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Layout-children-text
      </p>
    </main>
  </div>
</div>
`;
